sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4059,6 +4075,11 @@ pi_result piEnqueueEventsWaitWithBarrier(pi_queue Queue,
   PI_ASSERT(Queue, PI_INVALID_QUEUE);
   PI_ASSERT(Event, PI_INVALID_EVENT);
 
+  _pi_ze_event_list_t TmpWaitList;
```

Comment:
NOTE: there is a similar change to this function coming in https://github.com/intel/llvm/pull/3045
I think I prefer to wait for this PR to land, and rebase #3045 over it, so please go ahead.